app.js

```diff
@@ -2,6 +2,16 @@ const express = require("express");
 const { getAllTopics } = require("./controllers/topics.controller");
 const app = express();
 
+app.use((req, res, next) => {
+    const path = req.url.split("/");
+
+    if (path.includes("topics")) {
+        next();
+    } else {
+        res.status(404).send({ msg: "Not found!" });
+    }
+});
```

Comment:
Hey Hassen. In this case, the error handling would pick up anything that doesn't include "topics" in the path. So for future tickets, like "api/articles" for example, this would result in a 404 error which we wouldn't want. 
Could you amend this code please to make it dynamic?

Reply:
Hi, I am a confused by for error handling for this endPoint? How do I come up with it and where do I put and check the incoming request path? Thank you so much!

Reply:
can I start the next question? I thought express can handle this by default.

Reply:
Hey Hassen, I've just sent this message on Slack, but I'm just putting it here again for reference....

Express can handle a 404 by default, so you can choose to remove the 404 code you've written if you wish and move on to the next ticket.
It's just that the way it's written at the moment wouldn't work for any future endpoints because of how it's written. 👍   Let me know if you have any further questions about it